WordPress/gutenberg
# Query Pagination: Don't render an empty container

## Description
Fixes #30522.
Alternative to #33635.

Updates Query Pagination block to use `render_callback` for rendering container `div` element. Render callback won't render anything if block content is empty.

## How has this been tested?
1. Add Query block with pagination to page or template.
2. Change the `perPage` number so it shouldn't display pagination.
3. Verify that no empty pagination container is rendered.
4. Change the `perPage` once again, so now it renders paginations.
5. Verify that pagination works correctly.

### Example Query
```

<div class="wp-block-query">





<hr class="wp-block-separator"/>






<div class="wp-block-query-pagination">



</div>
</div>

```

## Types of changes
Bugfix

## Checklist:
